prov/rxm/src/rxm_ep.c

```diff
@@ -512,13 +516,28 @@ static ssize_t rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov
 		pkt_size = sizeof(*pkt) + pkt->hdr.size;
 	}
 
+	if (flags & FI_INJECT & ~FI_COMPLETION) {
```

Comment:
I think this should be ((flags & FI_INJECT) && !(flags & FI_COMPLETION)). flags & FI_INJECT would zero out FI_COMPLETION bit if present.